plugin/src/main/kotlin/nl/avisi/anowalke/toolWindow/FluffyListener.kt

```diff
@@ -42,42 +47,89 @@ class FluffyListener(val project: Project) : ToolWindowManagerListener {
         currentClass.text = FileChooser.chooseFiles(fileChooserDescriptor, project, null).first().name
     }
 
-    fun scan() {
-        scanProject()
+    fun scan(result: JTextArea) {
+        log.warn("Initiate scan function.")
+        val identifiers = scanProject()
+        val stemmedIdentifiers: List<Identifier> = identifiers.map {
+            Identifier(it.className, stemmer.stem(it.expression))
+        }
+        stemmedIdentifiers.forEach { log.warn("Stemmed identifier: $it") }
+        val results = findShotgunSurgery(stemmedIdentifiers)
+        result.text = results.toString()
     }
 
-    private fun getJavaFiles(file: VirtualFile): List<String>? {
+    private fun getJavaFiles(file: VirtualFile): List<Identifier>? {
+        log.warn("Initiate getJavaFiles function.")
         // Filter out hidden files
         if (!file.isDirectory && !file.name.startsWith(".")) {
-            PsiManager.getInstance(project).findFile(file)?.let { it -> log.warn(it.name) } ?: log.warn("no file found")
-            val expressions = PsiManager.getInstance(project).findFile(file)?.let { psiFile -> psiService.extract(psiFile) }
-//            log.warn("And now pretty")
-//            expressions?.sanatise()?.forEach { log.warn(it) }
-            return expressions
-        } else {
-            file.children.forEach { getJavaFiles(it) }
+            log.warn("File: ${file.name} is not a directory nor a hidden file.")
+            var expressions = PsiManager.getInstance(project).findFile(file)?.let { psiFile ->
+                psiService.extract(psiFile)
+            }
+            if (expressions != null) {
+                log.warn("Extract function retrieved ${expressions.size} expressions.")
+                expressions = sanitiser.sanitise(expressions)
+            }
+            log.warn("Sanitise function retrieved ${expressions?.size} expressions.")
+            return expressions?.map { expression ->
```

Comment:
Kan je dit niet in bovenstaande if opnemen. Daar weet je dat expressions niet null is.